tests/cancun/eip4788_beacon_root/conftest.py

```diff
@@ -0,0 +1,181 @@
+
+
+@pytest.fixture
+def env(timestamp: int, beacon_root: bytes) -> Environment:  # noqa: D103
+    return Environment(
+        timestamp=timestamp,
+        parent_beacon_block_root=beacon_root,
+    )
+
+
+@pytest.fixture
+def call_type() -> Op:  # noqa: D103
+    return Op.CALL
+
+
+@pytest.fixture
+def call_gas() -> int:  # noqa: D103
+    return BEACON_ROOT_PRECOMPILE_GAS
+
+
+@pytest.fixture
+def call_address() -> int:  # noqa: D103
+    return to_address(0x100)
+
+
+@pytest.fixture
+def precompile_call_account(call_type: Op, call_gas: int) -> Account:
+    """
+    Code to call the beacon root precompile.
+    """
+    precompile_call_code = Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE)
+    if call_type == Op.CALL or call_type == Op.CALLCODE:
+        precompile_call_code += Op.SSTORE(
+            0,  # store the result of the precompile call in storage[0]
+            call_type(
+                call_gas,
+                BEACON_ROOT_PRECOMPILE_ADDRESS,
+                0x00,
+                0x00,
+                Op.CALLDATASIZE,
+                0x00,
+                0x20,
+            ),
+        )
+    elif call_type == Op.DELEGATECALL or call_type == Op.STATICCALL:
+        # delegatecall and staticcall use one less argument
+        precompile_call_code += Op.SSTORE(
+            0,
+            call_type(
+                call_gas,
+                BEACON_ROOT_PRECOMPILE_ADDRESS,
+                0x00,
+                Op.CALLDATASIZE,
+                0x00,
+                0x20,
+            ),
+        )
+    precompile_call_code += (
+        Op.SSTORE(1, Op.MLOAD(0x00))
+        + Op.SSTORE(2, Op.RETURNDATASIZE)
+        + Op.RETURNDATACOPY(0, 0x0, Op.RETURNDATASIZE)
+        + Op.SSTORE(3, Op.MLOAD(0x00))
+    )
+    return Account(
+        nonce=0,
+        code=precompile_call_code,
+        balance=10**10,
+    )
+
+
+@pytest.fixture
+def valid_call() -> bool:
+    """
+    Validity of beacon root precompile call: defaults to True.
+    """
+    return True
+
+
+@pytest.fixture
+def valid_input() -> bool:
+    """
+    Validity of timestamp input to precompile call: defaults to True.
+    """
+    return True
+
+
+@pytest.fixture
+def pre(
+    precompile_call_account: Account,
+    call_address: str,
+) -> Dict:
+    """
+    Prepares the pre state of all test cases, by setting the balance of the
+    source account of all test transactions, and the precompile caller account.
+    """
+    return {
+        TestAddress: Account(
+            nonce=0,
+            balance=0x10**10,
+        ),
+        call_address: precompile_call_account,
+    }
+
+
+@pytest.fixture
+def tx(
+    call_address: str,
+    timestamp: int,
+) -> Transaction:
+    """
+    Prepares transaction to call the beacon root precompile caller account.
+    """
+    return Transaction(
+        ty=2,
+        nonce=0,
+        data=to_hash_bytes(timestamp_index(timestamp)),
```

Comment:
We might need to add some tests where the data is not bounded to 32-bytes (result from `to_hash_bytes`), to test pre-compile inputs with 33+ bytes (should fail) and <32 bytes (should be padded).